ruby-challenges.rb

```diff
@@ -11,6 +11,43 @@
 # Expected output: ['tea', 'water', 'soda water']
 
 # Pseudo code:
+#       input: array of words ['coffee', 'tea', 'juice', 'water', 'soda water']
+#       output: array of words only containing a particular letter 
+#               'o', ['coffee', 'soda water']
+#               't', ['tea', 'water', 'soda water']
+#       process: take in array
+#                iterate 
+#                create a method and search through the array for the matching words, .include?()
+#                .each method which takes a block and the parameter of value ||
+#                keyword 'if' letter is in words return only those words
+
+
+# This was my first attempt at the problem, definitly not right and missing many puzzle pieces
+# def part_letter (array) 
+#     array.map do |value|
+#         if value.include?('o')
+#             []
+#         elsif value.inlcude?('t')
+#             []
+#         else 
+#             'something went wrong'
+#         end
+#     end 
+# end
+
+# then tried this, feels more right but still not quite there
+# def part_letter(array, letter)
+#     array.each do |value|
+#         if value.include?(letter)
+#             puts value
+#         end 
+#     end
+# end
+# part_letter(filter_letter_o)
+# part_letter(filter_letter_t)
+#output: wrong number of arguments (given 1, expected 2)
```

Comment:
Look at what your functions is taking in. Remember, for every parameter there is a corresponding argument 😊